internal/serve/httphandler/login_handler.go

```diff
@@ -49,13 +50,18 @@ type LoginHandler struct {
 	ReCAPTCHAValidator validators.ReCAPTCHAValidator
 	MessengerClient    message.MessengerClient
 	Models             *data.Models
-	ReCAPTCHAEnabled   bool
-	MFAEnabled         bool
 }
 
 func (h LoginHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	ctx := req.Context()
 
+	tnt, tntErr := tenant.GetTenantFromContext(ctx)
+	if tntErr != nil {
+		log.Ctx(ctx).Errorf("error getting tenant from context: %s", tntErr)
+		httperror.Unauthorized("", tntErr, nil).Render(rw)
```

Comment:
do you mind explaining to me why this would be a 401 and not 404?

Reply:
I thought it would be more accurate to return an unauthorized error if we could not get the tenant from the context during the login process.

Reply:
ok, got it! my thinking was that if the user couldn't log in, it's already an implicit `unauthorized` error and `not found` could be more diagnostic for the actual error which is we cannot find the tenant the context is referring to.

Reply:
Good point, @ziyliu - I've no strong opinion on this.